tests/testthat/test-calculate_age.R

```diff
@@ -0,0 +1,98 @@
+data <- readRDS(system.file("extdata", "test_df.RDS", package = "cleanepi"))
+
+test_that("calculate_age works", {
+  age <- calculate_age(
+    data               = data,
+    target_column      = "dateOfBirth",
+    end_date           = Sys.Date(),
+    age_in             = "months",
+    age_remainder_unit = "days",
+    na_strings         = "-99"
+  )
+  expect_s3_class(age, "data.frame")
+  expect_identical(ncol(age), 10L)
+  expect_identical(names(age)[9L:10L], c("age_in_months", "remainder_days"))
+})
+
+test_that("calculate_age works", {
+  age <- calculate_age(
+    data               = data,
+    target_column      = "dateOfBirth",
+    end_date           = Sys.Date(),
+    age_in             = "months",
+    age_remainder_unit = "weeks",
+    na_strings         = "-99"
+  )
+  expect_s3_class(age, "data.frame")
+  expect_identical(ncol(age), 10L)
+  expect_identical(names(age)[9L:10L], c("age_in_months", "remainder_weeks"))
+})
+
+test_that("calculate_age works", {
+  age <- calculate_age(
+    data               = data,
+    target_column      = "dateOfBirth",
+    end_date           = Sys.Date(),
+    age_in             = "months",
+    age_remainder_unit = "weeks",
+    na_strings         = "-99"
+  )
+  expect_s3_class(age, "data.frame")
+  expect_identical(ncol(age), 10L)
+  expect_identical(names(age)[9L:10L], c("age_in_months", "remainder_weeks"))
+})
+
+test_that("calculate_age works", {
+  age <- calculate_age(
+    data               = data,
+    target_column      = "dateOfBirth",
+    end_date           = Sys.Date(),
+    age_in             = "years",
+    age_remainder_unit = "months",
+    na_strings         = "-99"
+  )
+  expect_s3_class(age, "data.frame")
+  expect_identical(ncol(age), 10L)
+  expect_identical(names(age)[9L:10L], c("age_in_years", "remainder_months"))
+})
+
+test_that("calculate_age works", {
+  age <- calculate_age(
+    data               = data,
+    target_column      = "dateOfBirth",
+    end_date           = Sys.Date(),
```

Comment:
Could you also add tests where the `end_date` is provided in different formats (some bad formats) and expect errors/warnings where appropriate? `Sys.Date()` is always properly formatted.

Reply:
will work on this - Thanks @jamesmbaazam